### PR TITLE
[formula: natalie] Compile natalie.swift for performance

### PR DESCRIPTION
**Pull request by ldiqual**
This commit adds the compilation of `natalie.swift` during the installation process, resulting in a greatly improved execution time. This is very important because when `natalie` is part of an app build process, having a fast build phase for generating Storyboard code greatly improves build times.

Using the original `natalie.swift` script:
```
$ time ./natalie.swift NatalieExample/
// Output
3.04 real         2.85 user         0.12 sys
```

Using a compiled version:
```
$ time natalie NatalieExample/
// Output
0.07 real         0.01 user         0.03 sys
```

@krzyzanowskim (original author of the program): Let me know what you think.

### CODE REVIEW

File: Library/Formula/natalie.rb [hunk 1]
```diff
@@ -8,14 +8,15 @@ class Natalie < Formula
   depends_on :xcode => "6.3"
 
   def install
-    bin.install "natalie.swift"
+    system "xcrun -sdk macosx swiftc -O natalie.swift -o natalie"
```
Comment (tdsmith) [inline Library/Formula/natalie.rb]: "split", "like", "this" or use `%W[]`, please

File: Library/Formula/natalie.rb [hunk 2]
```diff
@@ -8,14 +8,15 @@ class Natalie < Formula
...
+    bin.install "natalie"
```
Comment (tdsmith) [inline Library/Formula/natalie.rb]: let's also install `natalie.swift` so we don't break installed uses

Comment (tdsmith) [inline Library/Formula/natalie.rb]: (you can pass multiple arguments to install, like `bin.install "natalie", "natalie.swift"`)

**Comment by krzyzanowskim** [comment]
Ok. Just please keep both versions installed (as suggested by @tdsmith)
"natalie" <- compiled
"natalie.swif" <- script version

**Comment by mikemcquaid** [comment]
@krzyzanowskim What's the advantage of running the slower version?

**Comment by krzyzanowskim** [comment]
@mikemcquaid surely that compiled version is faster, on the other hand this is still a script, and I want to keep it this way (at least for a while). It thought it may be good idea to have it installed this way for consistency with the current state of the project (and docs). 

Now I think that maybe nobody will ever check if executed file is binary or script so maybe not a big deal at all.

Anyway, executable shouldn't be renamed to "natalie".

**Comment by mikemcquaid** [comment]
@krzyzanowskim Up to you which but I'd really rather we picked only one to install into `bin`. The other could always go in `libexec` or `share` or something.

**Comment by dunn** [comment]
> Anyway, executable shouldn't be renamed to "natalie".

Why not?  It's the name of the formula, so it seems like the most natural thing IMO.

**Comment by krzyzanowskim** [comment]
@dunn that would introduce unnecessary complication: why path is different when installed with brew. brew is to make the process easier, not harder.

**Comment by krzyzanowskim** [comment]
ok, pls go with compiled bin/natalie.swift 

btw @ldiqual check this https://github.com/krzyzanowskim/Natalie/issues/60 for performance tip.

**Comment by ldiqual** [comment]
@krzyzanowskim @tdsmith @mikemcquaid Fixed.

@mikemcquaid I tried `system "xcrun", "-sdk macosx", "swiftc", "-O", "natalie-script.swift", "-o natalie.swift"` but it failed with the following error:

```
==> xcrun -sdk macosx swiftc -O natalie-script.swift -o natalie.swift
  -k, --kill-cache            invalidate all existing cache entries
  --show-sdk-path             show selected SDK install path
  --show-sdk-version          show selected SDK version
  --show-sdk-platform-path    show selected SDK platform path
  --show-sdk-platform-version show selected SDK platform version
```

So it seems that `xcrun` doesn't support the way the arguments are passed behind the scenes, any idea why?

**Comment by tdsmith** [comment]
Keep splitting: `system "xcrun", "-sdk", "macosx", "swiftc", "-O", "natalie-script.swift", "-o", "natalie.swift"` -- you have to do all the tokenization that the shell would normally do for you, and the shell splits on whitespace unless it's quoted

**Comment by ldiqual** [comment]
@tdsmith Fixed. Just for my own curiosity: having a long string worked and it seemed clearer to me, but I'm sure there is a reason why it's better to do it your way. Would you mind linking me to a previous discussion or elaborate on why it's better? Again, just curious.

**Comment by mikemcquaid** [comment]
> @tdsmith Fixed. Just for my own curiosity: having a long string worked and it seemed clearer to me, but I'm sure there is a reason why it's better to do it your way. Would you mind linking me to a previous discussion or elaborate on why it's better? Again, just curious.

We prefer that style because it makes it doesn't create an unnecessary subshell and prevents accidental arbitrary command execution.

File: Library/Formula/natalie.rb [hunk 3]
```diff
@@ -8,6 +8,8 @@ class Natalie < Formula
   depends_on :xcode => "6.3"
 
   def install
+    system "mv", "natalie.swift", "natalie-script.swift"
```
Comment (DomT4) [inline Library/Formula/natalie.rb]: Shouldn't need to call `system` on `mv`.

Comment (mikemcquaid) [inline Library/Formula/natalie.rb]: Ping.

**Comment by ldiqual** [comment]
@mikemcquaid Thanks for the explanations!

@DomT4 Fixed

**Comment by dunn** [comment]
Thanks! Merged in 1220934c32742f4496c9728fc213dffe90f98071.

**Pull request closed without merging**